packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -37,10 +37,16 @@ class CreateCommand extends FlutterCommand {
       help: 'Also add a flutter_driver dependency and generate a sample \'flutter drive\' test.'
     );
     argParser.addFlag(
-      'plugin',
+      'package',
```

Comment:
I think we should switch to using an option to select project types; something like `--template=<type>`, where type is one of `application`, `package`, or `plugin`. That feels a bit cleaner to me than having several different, mutually exclusive, cli flags.